doc/modules/mod_pubsub.md

```diff
@@ -0,0 +1,67 @@
+### What is PubSub?
+
+PubSub is a fascinating design pattern which mostly promotes a loose coupling between two kinds of entity (publishers and subscribers).
+Like their names suggest, in the pubsub world we have publishers who fire events, and subscribers who wish to be notified about those events when publishers push data.
+There might be several subscribers, several publishers, and even several channels (or nodes) where the events are sent.
+
+### Module Description
+
+This module implements [XEP-0060 (Publish-Subscribe)](http://www.xmpp.org/extensions/xep-0060.html).
+Due to the complexity of the protocol, the PubSub engine makes successive calls to `nodetree` and `node plugins` in order to check the validity of requests, perform the corresponding action and return a result or appropriate error.
+Such an architecture makes it much easier to write custom pubsub plugins and add new storage backends.
+It's all about fitting PubSub to your needs!
+
+### Options
+
+* `host` (string, default: `"pubsub.@HOST@"`): Subdomain for Pubsub service to reside under.
+`@HOST@` is replaced with each served domain.
+* `access_create` (atom, default: `all`): Who is allowed to create pubsub nodes.
+* `max_items_node` (integer, default: `10`): Define the maximum number of items that can be stored in a node.
+* `max_subscriptions_node` (integer, default: `undefined` - no limitation): The maximum number of subscriptions managed by a node.
+* `nodetree` (binary, default: `<<"tree">>`): Specify the storage and organisation of the pubsub nodes.
+Called on `get`, `create` and `delete` node.
+Only one nodetree can be used per host, and is shared by all node plugins.
+    * `<<"tree">>"` - store nodes in database (only mnesia supported).
+    * `<<"virtual">>` - does not store nodes in database.
+    This saves resources on systems when the number of nodes is large.
+    If using the `<<“virtual”>>` nodetree, you can only enable these node plugins:
+    `[<<“flat”>>,<<“pep”>>]` or `[<<“flat”>>]`; any other plugin configuration will not work.
+    Also, all nodes will have the defaut configuration, and this can not be changed.
+    Using the `<<“virtual”>>` nodetree requires the pubsub module to be started with a clean database, it will not work if you used the default `<<“tree”>>` nodetree before.
+    * `<<"dag">>"` - provides experimental support for [XEP-0248 (PubSub Collection Nodes)](http://xmpp.org/extensions/xep-0248.html).
+    In this case you should also add the `<<“dag”>>` node plugin as default, for example: plugins: `[<<"dag">>,<<"flat">>,<<"hometree">>,<<"pep">>]`
+* `ignore_pep_from_offline` (boolean, default: `true`): specify whether or not we should get last published PEP items from users in our roster which are offline when we connect.
+The default option is `true` hence we will get only the last items from the online contacts.
+* `last_item_cache` (boolean, default `false`): specify whether or not pubsub should cache the last items.
```

Comment:
The main reason for `last_item_cache` is the ability to send the last published item to a new subscriber.